### PR TITLE
BugFix: Variable names with multiple underscores invalid in ui when creating a new variable

### PR DESCRIPTION
**Pull request by pleek91**
# Description
Updates all the components and compositions used in the dashboard to accepts and pass getters rather than refs for filters. This way the dashboard filters can update with each polling interval so the 24h time span constantly updates.

### CODE REVIEW

File: src/components/DashboardWorkPoolFlowRunCompleteness.vue
Comment (pleek91) [inline src/components/DashboardWorkPoolFlowRunCompleteness.vue:1]: This isn't completely new. Git didn't follow the rename well.

**Comment by pleek91** [comment]
@znicholasbrown please let me know if this is to large and I can break it up into separate PRs per card.

File: src/compositions/useTaskRunsCount.ts [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 import { TaskRunsFilter } from '@/models/Filters'
 import { WorkspaceTaskRunsApi } from '@/services/WorkspaceTaskRunsApi'
 import { Getter } from '@/types/reactivity'
 import { UseEntitySubscription } from '@/types/useEntitySubscription'
 
-export type UseTaskRunsCount = UseEntitySubscription<WorkspaceTaskRunsApi['getTaskRunsCount'], 'taskRunsCount'>
+export type UseTaskRunsCount = UseEntitySubscription<WorkspaceTaskRunsApi['getTaskRunsCount'], 'count'>
 
-export function useTaskRunsCount(flowRunId: MaybeRefOrGetter<string | null | undefined>): UseTaskRunsCount {
+export function useTaskRunsCount(filter: MaybeRefOrGetter<TaskRunsFilter | null | undefined>, options?: SubscriptionOptions): UseTaskRunsCount {
```
Comment (pleek91) [inline src/compositions/useTaskRunsCount.ts:12]: Technically this is a breaking change. However this composition is yet to be used outside of ui-library so this can be released without breaking oss or cloud.

File: src/compositions/useTaskRunsHistory.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { SubscriptionOptions, UseSubscription, useSubscriptionWithDependencies } from '@prefecthq/vue-compositions'
+import merge from 'lodash.merge'
+import { computed, ComputedRef, MaybeRefOrGetter, toRef, toValue } from 'vue'
+import { TaskRunHistory } from '..'
```
Comment (znicholasbrown) [inline src/compositions/useTaskRunsHistory.ts:4]:
```suggestion
import { TaskRunHistory } from '@/models'
```
I think this is the correct import but I'm not totally sure

File: src/compositions/useTaskRunsCount.ts [hunk 3]
```diff
@@ -17,30 +18,30 @@ export function useTaskRunsCount(flowRunId: MaybeRefOrGetter<string | null | und
...
   const parameters = toRef(getter)
-  const subscription = useSubscriptionWithDependencies(api.taskRuns.getTaskRunsCount, parameters)
-  const taskRunsCount = computed(() => subscription.response)
+  const subscription = useSubscriptionWithDependencies(api.taskRuns.getTaskRunsCount, parameters, options)
+  const count = computed(() => subscription.response)
 
   return {
     subscription,
-    taskRunsCount,
+    count,
```
Comment (znicholasbrown) [inline src/compositions/useTaskRunsCount.ts:45]: ❤️

File: src/components/WorkPoolAverageLateTime.vue [hunk 4]
```diff
@@ -1,39 +1,41 @@
...
+  const lateFlowRunsFilter: Getter<FlowRunsFilter> = () => {
+    const filter: FlowRunsFilter = {
+      workPools: {
+        name: [props.workPool.name],
       },
-    },
-  }))
+      flowRuns: {
+        state: {
+          name: ['Late'],
```
Comment (znicholasbrown) [inline src/components/WorkPoolAverageLateTime.vue:30]: Am I correct in assuming the API accepts and we have use cases for passing flow run states that _aren't_ late to the average lateness composition? If not, it might make sense to bake this into the composition

File: src/components/PageHeadingFlowRun.vue [hunk 5]
```diff
@@ -85,12 +83,15 @@
...
+  const { flowRun } = useFlowRun(() => props.flowRunId, { interval: 30000 })
 
   const isPending = computed(() => flowRun.value?.stateType ? isPendingStateType(flowRun.value.stateType) : true)
-  const { taskRunsCount } = useTaskRunsCount(flowRunId)
+
+  const { count: taskRunsCount } = useTaskRunsCount(() => ({
+    flowRuns: {
+      id: [props.flowRunId],
+    },
+  }))
```
Comment (znicholasbrown) [inline src/components/PageHeadingFlowRun.vue:94]: The anonymous function makes this a little tougher to read than I'd hope (the one on line 86 is fine)

File: src/components/FlowRunsAccordion.vue [hunk 6]
```diff
@@ -2,41 +2,42 @@
...
   import { toMap } from '@/utilities'
 
   const props = defineProps<{
-    filter: FlowRunsFilter,
+    filter: MaybeGetter<FlowRunsFilter>,
   }>()
 
-  const flowsFilter = computed<FlowsFilter>(() => {
+  const flowRunsFilter = toRef(props.filter)
+  const flowsFilterGetter: Getter<FlowsFilter> = () => {
```
Comment (znicholasbrown) [inline src/components/FlowRunsAccordion.vue:30]: I'm not sure about this rename - we don't use variableGetter elsewhere and I'm not sure it's necessary. For consistency I'd prefer we retained `flowsFilter` instead

File: src/components/FlowRunStateTypeTabs.vue [hunk 7]
```diff
@@ -54,19 +56,19 @@
...
-    const types = getTabStates(tab)
-
-    return {
-      ...props.filter,
-      flowRuns: {
-        ...props.filter?.flowRuns,
-        state: {
-          ...props.filter?.flowRuns?.state,
-          type: types,
+  function getStateTypeFilterGetter(tab: string): Getter<FlowRunsFilter> {
```
Comment (znicholasbrown) [inline src/components/FlowRunStateTypeTabs.vue:59]: And in direct conflict with my other comment, I think this name change is fine since it's describing the return value lol

File: src/components/FlowRunListItem.vue [hunk 8]
```diff
@@ -65,8 +65,11 @@
   const tags = computed(() => props.flowRun.tags)
   const value = computed(() => props.flowRun.id)
 
-  const flowRunId = computed(() => props.flowRun.id)
-  const { taskRunsCount } = useTaskRunsCount(flowRunId)
+  const { count: taskRunsCount } = useTaskRunsCount(() => ({
```
Comment (znicholasbrown) [inline src/components/FlowRunListItem.vue:68]: Same here wrt anonymous function not helping readability

**Review by znicholasbrown — APPROVED** [review]
Added a few comments wrt passing anonymous functions that return nested objects but overall I don't have any concerns with the changes

File: src/components/PageHeadingFlowRun.vue (hunk 5 above)
Comment (pleek91) [reply src/components/PageHeadingFlowRun.vue:94]: Personally for small filters like this I prefer this and find it reads nicely. But if it gets much larger than this, yeah I'd wanna break it out.

File: src/components/WorkPoolAverageLateTime.vue (hunk 4 above)
Comment (pleek91) [reply src/components/WorkPoolAverageLateTime.vue:30]: Hmm good catch. I'm actually not sure this is correct at all. But I'll check on it

File: src/components/PageHeadingFlowRun.vue (hunk 5 above)
Comment (znicholasbrown) [reply src/components/PageHeadingFlowRun.vue:94]: Ack - fine by me

File: src/components/FlowRunsAccordion.vue (hunk 6 above)
Comment (pleek91) [reply src/components/FlowRunsAccordion.vue:30]: yeah I don't disagree. Will update.

File: src/components/FlowRunStateTypeTabs.vue (hunk 7 above)
Comment (pleek91) [reply src/components/FlowRunStateTypeTabs.vue:59]: Lol same cause it returns a getter.

File: src/components/FlowRunListItem.vue (hunk 8 above)
Comment (pleek91) [reply src/components/FlowRunListItem.vue:68]: Talked about this in slack. In github it is kinda hard to parse. But in vscode it looks nice. For now we'll keep these but might revisit

image

File: src/components/WorkPoolAverageLateTime.vue (hunk 4 above)
Comment (pleek91) [reply src/components/WorkPoolAverageLateTime.vue]: This shouldn't even have the late state filter. So removed that. Thanks for calling that out!
